wgpu/src/backend/web.rs

```diff
@@ -1169,9 +1168,7 @@ impl crate::Context for Context {
             crate::ShaderSource::SpirV(ref spv) => {
```

Comment:
Now that `&str` is accepted, the `SpirV` variant won't compile anymore. Instead this should use naga to convert SPIR-V to WGSL, then use `WGSL` instead

Reply:
I've taken a first go at converting the spir-v data into a WGSL shader module desc. I'm currently unsure of what to do in regards to error handling, as `device_create_shader_module` doesn't return a result, however, just throwing around `unwrap` feels wrong.

Reply:
I forgot that we should be able to follow the same logic as `direct` here:
https://github.com/gfx-rs/wgpu/blob/master/wgpu/src/backend/direct.rs#L971-L978
The only difference is that at the end we would do `back::wgsl::write_string` instead of returning the naga module

The `unwrap` here is fine for now, we'll properly include this in error logging later

Reply:
Ah, that makes sense, and avoids the `unsafe` I had to use for converting [u32] to [u8]. Also gives me the preferred parse options, rather than `..Default::default()`, thanks.